pyclient/pydeephaven/table.py

```diff
@@ -0,0 +1,683 @@
+
+    def avg_by(self, column_names: List[str]) -> Table:
+        """ Perform avg-by aggregation on the table and return the result table. Columns not used in the grouping must
+        be of numeric types.
+
+
+        Args:
+            column_names (List[str]): the group-by column names
+
+        Returns:
+            a Table object
+
+        Raises:
+            DHError
+
+        """
+        return self.session.table_service.dedicated_aggregator(self, agg_type=AggType.AVG,
+                                                               group_by_columns=column_names)
+
+    def std_by(self, column_names: List[str]) -> Table:
+        """ Perform std-by aggregation on the table and return the result table. Columns not used in the grouping
+        must be of numeric types.
+
+
+        Args:
+            column_names (List[str]): the group-by column names
+
+        Returns:
+            a Table object
+
+        Raises:
+            DHError
+
+        """
+        return self.session.table_service.dedicated_aggregator(self, agg_type=AggType.STD,
+                                                               group_by_columns=column_names)
+
+    def var_by(self, column_names: List[str]) -> Table:
+        """ Perform var-by aggregation on the table and return the result table. Columns not used in the grouping
+        must be of numeric types.
+
+
+        Args:
+            column_names (List[str]): the group-by column names
+
+        Returns:
+            a Table object
+
+        Raises:
+            DHError
+
+        """
+        return self.session.table_service.dedicated_aggregator(self, agg_type=AggType.VAR,
+                                                               group_by_columns=column_names)
+
+    def median_by(self, column_names: List[str]) -> Table:
+        """ Perform median-by aggregation on the table and return the result table. Columns not used in the grouping
+        must be of numeric types.
+
+
+        Args:
+            column_names (List[str]): the group-by column names
+
+        Returns:
+            a Table object
+
+        Raises:
+            DHError
+
+        """
+        return self.session.table_service.dedicated_aggregator(self, agg_type=AggType.MEDIAN,
+                                                               group_by_columns=column_names)
+
+    def min_by(self, column_names: List[str]) -> Table:
+        """ Perform min-by aggregation on the table and return the result table. Columns not used in the grouping
+        must be of numeric types.
+
+
+        Args:
+            column_names (List[str]): the group-by column names
+
+        Returns:
+            a Table object
+
+        Raises:
+            DHError
+
+        """
+        return self.session.table_service.dedicated_aggregator(self, agg_type=AggType.MIN,
+                                                               group_by_columns=column_names)
+
+    def max_by(self, column_names: List[str]) -> Table:
```

Comment:
Empty list by default. A lot more too... I've listed a few, but I think all of the `_by` methods should have an empty list by default.